duplicate.js

```diff
@@ -0,0 +1,34 @@
+function duplicate(arr){
+  var a =new Set(arr);
```

Comment:
Again a miss read of the problem statement `Try to work within the array without using any other temporary data structures.`

The set here ([poorly](https://blog.elpassion.com/naming-101-quick-guide-on-how-to-name-things-a9bcbfd02f02) named a) is a temporary data structure.